app/src/main/java/sdp/moneyrun/player/LocalPlayer.java

```diff
@@ -50,22 +56,32 @@ public void updateLostCoins(ArrayList<Coin> availableCoins){
         this.lostCoins.removeAll(toRemove);
     }
 
-    public void updateLocallyAvailableCoins(ArrayList<Coin> availableCoins, boolean locally, Coin toRemove){
-
-        if(!locally){
-
-            if(availableCoins == null){
+    /**
+     * This is the function that will do all the work, if the locally parameter is set to true it will only remove the given coin from
+     * the locallyAvailableCoins and add it to the lostCoins. If locally is set to false, it will update both lists using the newly availableCoins
+     * list from the database.
+     *
+     * @param availableCoins The list of coins that are still available to EVERYONE
+     * @param locally Whether we are only removing a Coin locally or updating everything based on the list
+     *                of available coins from the dataBase.
+     * @param toRemove The coin to remove if locally set to true
+     */
+    public void updateLocallyAvailableCoins(ArrayList<Coin> availableCoins, boolean locally, Coin toRemove) {
```

Comment:
If I understand the function correctly, if you do locally = true,  you need to call the function with a null value for the first parameter ? (assuming you dont have access to availableCoins from wherever you are calling this from). If so, either explicitly specify in the javadoc that if you are doing it locally the first argument can be null, or separate it into two functions one for local one for not local. Just for clarity of whoever is calling this, so they know the first parameter can be null in certain cases.

Reply:
Done